tensorflow_addons/optimizers/discriminative_layer_training.py

```diff
@@ -0,0 +1,146 @@
+# Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Discriminative Layer Training Optimizer for TensorFlow."""
+
+import tensorflow as tf
+from typeguard import typechecked
+from typing import Union
+from tensorflow.keras.optimizers import Optimizer
+
+
+@tf.keras.utils.register_keras_serializable(package="Addons")
+class MultiOptimzer(Optimizer):
+    @typechecked
+    def __init__(
+        self,
+        optimizer_layer_pairs: Union[list, None] = None,
```

Comment:
I think `optimizers_and_layers` more fit the tensorflow naming philosophy like `apply_gradients(grads_and_vars)`.